app/src/main/java/org/oxycblt/auxio/util/ShareUtil.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023 Auxio Project
+ * ShareUtil.kt is part of Auxio.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+ 
+package org.oxycblt.auxio.util
+
+import android.content.Context
+import androidx.core.app.ShareCompat
+import org.oxycblt.auxio.music.Song
+
+private const val MIME_TYPE_FALLBACK = "audio/*"
+
+/**
+ * Show system share sheet to share song
+ *
+ * @param song the [Song] to share
+ */
+fun Context.shareSong(song: Song) {
```

Comment:
Generally I think I'd structure these methods as:

```
fun share(song: Song) = share(listOf(song))
fun share(songs: List<Song>) {
    // ...
}
```

Given the precedent set by classes like `MusicViewModel`. This is just a stylistic nit.